### ISSUE TITLE
HTTP2 stall on reset streams

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#3601]
We have a report that a client that is not consuming a servers HTTP2 stream content (until such time as the server TCP/IP flow control blocking), but then sends a reset frame, can result in a  stall.  The servers HTTP2Stream objects are stuck waiting in the HTTP2Flusher queue, but with remoteReset field as true.  This indicates that the reset has been received, but has not been sufficient to free those streams from the HTTP2Flusher - presumably due to TCP/IP flow control.

**Comment by alexjinghn** [comment]
Hi, just want to follow up on this. Since this issue is still OPEN, is there any progress in this investigation?

**Comment by sbordet** [comment]
See also #2184.

**Comment by sbordet** [comment]
@alexjinghn this is an exceedingly rare case, and I'll investigate this coming week.

**Comment by alexjinghn** [comment]
That's great to hear, thanks for the investigation!

**Comment by sbordet** [comment]
@alexjinghn I have pushed a fix in a PR. Would you be able to try it out?

**Comment by alexjinghn** [comment]
Thanks @sbordet , how do I get it? Is it already published somewhere?

**Comment by sbordet** [comment]
@alexjinghn it's not published, it's in a pull request branch:

```
$ git clone https://github.com/eclipse/jetty.project.git
$ cd jetty.project
$ git checkout jetty-9.4.x-3601-http2_stall_on_reset_stream
$ mvn clean install -DskipTests
```

**Comment by alexjinghn** [comment]
We use jetty in the embedded way, so it'll be hard if the jars are not published in maven.

If it's too much to ask, I plan to swap the jetty jars on the host after application is deployed. 

Is `./jetty.project/jetty-distribution/target/distribution/lib` the right place to look for those jars?

I.e. would jars under that path be the same jars published to maven should you choose to publish them?

**Comment by sbordet** [comment]
@alexjinghn I'm confused. The `jetty-distribution` directory is only used if you run Jetty as a standalone server.

If you use Jetty embedded, then you should have a Maven POM (or similar) where you declare your dependencies, and it suffices to update the version number to 9.4.20-SNAPSHOT and Maven does the rest.

Otherwise please be patient, we'll merge in the PR and deploy a snapshot in the next days.

**Comment by alexjinghn** [comment]
We are using customized gradle to build and such changes can't easily go in without us officially importing them in our repo. We will wait for the PR merge, please keep us updated.

**Comment by alexjinghn** [comment]
@sbordet I see that this PR is included in Jetty release 9.4.20, do you have a timeline for when 9.4.20 will be officially released?

**Comment by sbordet** [comment]
@alexjinghn yes it's included in 9.4.20 and we are finalizing the release today or tomorrow. It will take a few days to build, stage, test the release so expect next week (if all goes well) the 9.4.20 release to go public.

**Comment by alexjinghn** [comment]
Thanks! Eagerly waiting to test it.
